src/embind/embind_shared.js

```diff
@@ -220,11 +242,11 @@ var LibraryEmbindShared = {
     argsList = argsList.join(',')
     argsListWired = argsListWired.join(',')
 
-    var invokerFnBody = `
-      return function (${argsList}) {
-      if (arguments.length !== ${argCount}) {
-        throwBindingError('function ' + humanName + ' called with ' + arguments.length + ' arguments, expected ${argCount}');
-      }`;
+    var invokerFnBody = `return function (${argsList}) {\n`;
```

Comment:
As a followup can this be arrow function in some cases?

Reply:
I'll look into it. This is also used in constructors so having the scope bound, may be an issue.